File: packages/core/src/__tests__/steps/large-data-step.ts
import type { ApiResponse, ApiRouteConfig, ApiRouteHandler } from '../../types'

export const config: ApiRouteConfig = {
  type: 'api',
  name: 'large-data-step',
  emits: [],
  path: '/large-data',
  method: 'POST',
}

export const handler: ApiRouteHandler<string, ApiResponse<number, unknown>> = async (body) => {
  if (typeof body === 'string')
    return {
      status: 200,
      body: {
        'return data': 'random',
      },
    }
  if (Buffer.isBuffer(body)) return body.byteLength
  if (body && typeof (body as any).length === 'number') return (body as any).length
  return 0
}

File: packages/core/src/__tests__/steps/large-data-step.rb
def config
  {
    type: "api",
    name: "large-data-step",
    emits: [],
    path: "/large-data",
    method: "POST"
  }
end

def handler(body, _ctx)
  if body.is_a?(String)
    return {
      "status" => 200,
      "body" => { "return data" => "random" }
    }
  end

  if body.respond_to?(:type) && body.type == "Buffer" && body.respond_to?(:data)
    data = body.data
    return data.length if data.respond_to?(:length)
  end

  if body.respond_to?(:length)
    begin
      return body.length
    rescue StandardError
    end
  end

  0
end
